tutorials/run-laravel-on-appengine-standard/index.md

```diff
@@ -89,19 +114,154 @@ from laravel.com.
+            && !file_exists('/tmp/.dirs_created')) {
+            foreach (['/tmp/app/public',
+                        '/tmp/framework/cache/data',
+                        '/tmp/framework/sessions',
+                        '/tmp/framework/testing',
+                        '/tmp/framework/views',
+                        '/tmp/logs'] as $tmpdir) {
+                if (!file_exists($tmpdir)) {
+                    mkdir($tmpdir, 0755, true);
+                }
+            }
+            touch('/tmp/.dirs_created');
+        }
+        # [END]
+    
+    This code is where the `GCP_APP_ENGINE_LARAVEL` environment variable from 
+    `app.yaml` comes into play. This avoids running this code in local 
+    development environment.
+
+    If your code writes into any storage directory, you should make sure to 
+    create the necessary directory structure here.
+
+1.  Modify `composer.json` file by adding a few scripts scripts. To do this 
+    automatically, use the following commands:
+
+        composer config scripts.post-autoload-dump "Illuminate\Foundation\ComposerScripts::postAutoloadDump" "mkdir -p bootstrap/cache" "@php artisan package:discover --ansi"
+        composer config scripts.gcp-build "sed -i -e \"s|env('APP_STORAGE', base_path() . '/storage')|'/tmp'|g\" bootstrap/app.php && mkdir -p /tmp/framework/views && mv .env.gae .env && php artisan config:cache && rm -f .env && php artisan route:cache"
+
+    In the end your `"scripts"` section should look similar to this:
+
+        "scripts": {
+            "post-autoload-dump": [
+                "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
+                "mkdir -p bootstrap/cache",
+                "@php artisan package:discover --ansi"
+            ],
+            "post-root-package-install": [
+                "@php -r \"file_exists('.env') || copy('.env.example', '.env');\""
+            ],
+            "post-create-project-cmd": [
+                "@php artisan key:generate --ansi"
+            ],
+            "gcp-build": "sed -i -e \"s|env('APP_STORAGE', base_path() . '/storage')|'/tmp'|g\" bootstrap/app.php && mkdir -p /tmp/framework/views && mv .env.gae .env && php artisan config:cache && rm -f .env && php artisan route:cache"
+        }
+    
+    In `"scripts"` -> `"post-autoload-dump"` we add the following script right 
+    before `package:discover` command to ensure that the package discovery 
+    command has the directory to write into:
+
+        ...
+        "mkdir -p bootstrap/cache",
+        ...
     
-        composer remove --dev facade/ignition
+    In `"scripts"` we add the `"gcp-build"` field, which is a (currently 
+    undocumented) way of running custom logic during deployment to App Engine.
+
+        ...
+        "gcp-build": "sed -i -e \"s|env('APP_STORAGE', base_path() . '/storage')|'/tmp'|g\" bootstrap/app.php && mkdir -p /tmp/framework/views && mv .env.gae .env && php artisan config:cache && rm -f .env && php artisan route:cache"
```

Comment:
Possible improvement: instead of using sed to find and replace `env('APP_STORAGE', base_path() . '/storage')`, I think I might have found a better way. Using `env()` outside of configuration files causes problems if `php artisan cache:clear` is called during deployment. In my case, the environment variable always returned empty. I was able to fix this by replacing `env('APP_STORAGE', base_path() . '/storage')` with the following code in bootstrap/app.php:

```php
/*
|--------------------------------------------------------------------------
| Set Storage Path
|--------------------------------------------------------------------------
|
| This script allows you to override the default storage location used by
| the  application.  You may set the APP_STORAGE environment variable
| in your .env file,  if not set the default location will be used
|
*/
$app->useStoragePath($_ENV['APP_STORAGE'] ?: base_path() . '/storage');
```
This code uses the value of the environment variable if it is truthy, or falls back to a default if it is falsey (e.g. an empty string). This side-steps the problems caused by calling `env()` outside of a configuration file and, in my case, meant that I didn't have to use `sed` to find and replace code on the fly.